embassy-stm32/src/timer/pwm_input.rs

```diff
@@ -0,0 +1,134 @@
+//! Input capture driver.
+
+use embassy_hal_internal::into_ref;
+
+use super::low_level::{CountingMode, InputCaptureMode, InputTISelection, Timer};
+use super::{Channel, Channel1Pin, Channel2Pin, GeneralInstance4Channel};
+use crate::gpio::{AFType, Pull};
+use crate::time::Hertz;
+use crate::Peripheral;
+
+/// Input capture driver.
+pub struct PwmInput<'d, T: GeneralInstance4Channel> {
+    channel: Channel,
+    inner: Timer<'d, T>,
+}
+
+/// Convert pointer to TIM instance to TimGp16 object
+fn regs_gp16(ptr: *mut ()) -> crate::pac::timer::TimGp16 {
+    unsafe { crate::pac::timer::TimGp16::from_ptr(ptr) }
+}
+
+impl<'d, T: GeneralInstance4Channel> PwmInput<'d, T> {
+    /// Create a new input capture driver.
+    pub fn new(
+        tim: impl Peripheral<P = T> + 'd,
+        pin: impl Peripheral<P = impl Channel1Pin<T>> + 'd,
+        pull_type: Pull,
+        freq: Hertz,
+    ) -> Self {
+        into_ref!(pin);
+        critical_section::with(|_| {
+            pin.set_as_af_pull(pin.af_num(), AFType::Input, pull_type);
+            #[cfg(gpio_v2)]
+            pin.set_speed(crate::gpio::Speed::VeryHigh);
+        });
+
+        Self::new_inner(tim, freq, Channel::Ch1, Channel::Ch2)
+    }
+
+    /// Create a new input capture driver.
```

Comment:
```suggestion
    /// Create a new PWM input driver.
```
And it should be documented that in this case, the Channel 2 GPIO is the input of the signal if I understand correctly.